src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx

```diff
@@ -0,0 +1,269 @@
+
+  // the edit phone number button should not exist
+  expect(
+    view.queryByText(new RegExp(`edit.*${numberName}`, 'i'), {
+      selector: 'button',
+    }),
+  ).not.to.exist;
+  // and the phone number should not exist
+  // expect(view.queryByText(emailAddress)).not.to.exist;
+  // and the add phone number button should exist
+  expect(
+    view.getByText(new RegExp(`add.*${numberName}`, 'i'), {
+      selector: 'button',
+    }),
+  ).to.exist;
+}
+
+// When the update happens but not until after the Edit View has exited and the
+// user returned to the read-only view
+async function testSlowSuccess(numberName) {
+  server.use(...mocks.transactionPending);
+
+  const { phoneNumberInput } = deletePhoneNumber(numberName);
+
+  // wait for the edit mode to exit
+  await waitForElementToBeRemoved(phoneNumberInput);
+
+  // check that the "we're deleting your..." message appears
+  const deletingMessage = await view.findByText(
+    new RegExp(
+      `We’re in the process of deleting your ${numberName}. We’ll remove this information soon.`,
+      'i',
+    ),
+  );
+  expect(deletingMessage).to.exist;
+
+  server.use(...mocks.transactionSucceeded);
+
+  await waitForElementToBeRemoved(deletingMessage);
+
+  // the edit phone number button should not exist
+  expect(
+    view.queryByText(new RegExp(`edit.*${numberName}`, 'i'), {
+      selector: 'button',
+    }),
+  ).not.to.exist;
+  // and the phone number should not exist
+  // expect(view.queryByText(emailAddress)).not.to.exist;
+  // and the add phone number button should exist
+  expect(
+    view.getByText(new RegExp(`add.*${numberName}`, 'i'), {
+      selector: 'button',
+    }),
+  ).to.exist;
+}
+
+// When the initial transaction creation request fails
+async function testTransactionCreationFails(numberName) {
+  server.use(...mocks.createTransactionFailure);
+
+  deletePhoneNumber(numberName);
+
+  // expect an error to be shown
+  const alert = await view.findByTestId('edit-error-alert');
+  expect(alert).to.have.descendant('div.usa-alert-error');
+  // TODO: would be nice to be able to check the contents against a RegExp
+  expect(alert).to.contain.text('We’re sorry. We couldn’t update your');
+  expect(alert).to.contain.text('Please try again.');
+
+  // make sure that edit mode is not automatically exited
+  await wait(75);
+  expect(view.getByTestId('edit-error-alert')).to.exist;
+  const editButton = getEditButton();
+  expect(editButton).to.not.exist;
+}
+
+// When the update fails while the Edit View is still active
+async function testQuickFailure(numberName) {
+  server.use(...mocks.transactionFailed);
+
+  deletePhoneNumber(numberName);
+
+  // expect an error to be shown
+  const alert = await view.findByTestId('edit-error-alert');
+  expect(alert).to.have.descendant('div.usa-alert-error');
+  // TODO: would be nice to be able to check the contents against a RegExp
+  expect(alert).to.contain.text('We’re sorry. We couldn’t update your');
```

Comment:
Can we check here for `We’re sorry. We couldn’t update your ${numberName}` ?'

Reply:
Good call. Revised!